File: cloud/dynamic-site/gt-cs-1331-js/node/routes/solutions.js
const express = require("express");
const router = express.Router();
const fs = require("fs");
const vm = require("vm");
const console =require("console")
console.log("Hi")
console.log(__dirname)
const allSolutions = fs
  .readdirSync(__dirname + "/../solutions/")
  .map((name) => name.split(".")[0])
  .filter((name) => name !== "solution")
  .map((name) => eval(`new (require("../solutions/${name}"))`) /*{
    const code = fs.readFileSync(__dirname + "/../solutions/" + name + ".js", "utf-8");
    const sandbox={require,console,__dirname}
    const obj = vm.runInNewContext(code, sandbox);
    return new obj();
  }*/);

router.get("/", function (req, res, next) {
  res.send(
    allSolutions.map((solution) => {
      return {
        id: solution.id,
        label: solution.label,
        order: solution.orderNumber,
      };
    })
  );
});

router.get("/:id", function (req, res, next) {
  const sltn = allSolutions.filter(
    (solution) => solution.id === req.params.id
  )[0];
  res.send({ isPattern: sltn.isPattern, output: sltn.answer() });
});

module.exports = router;

File: cloud/dynamic-site/gt-cs-1331-py/app/solution_api/solutions.py
from collections import namedtuple

class Solutions:
    def __init__(self, module):
        names = filter(lambda name: name.startswith("Solution") and len(name) > 8, dir(module))
        slts = map(lambda nm: getattr(module, nm)(),names)
        self._solutions = list(slts)

    def list(self):
        list_item = namedtuple("ListItem","id label order_number")
        return list(map(lambda sltn: list_item(sltn.id, sltn.label, sltn.order_number), self._solutions))

    def get_solution(self, id):
        response = namedtuple("Response","label is_pattern answer")
        sltn = next(filter(lambda sltn: sltn.id==id, self._solutions))
        return response(sltn.label, sltn.is_pattern, sltn.answer)
